packages/itwin/ec3-widget/src/components/TemplateMenu.tsx

```diff
@@ -138,165 +123,141 @@ export const TemplateMenu = ({ template, goBack, created }: TemplateProps) => {
     return newReportOptions;
   }, [availableReports]);
 
-  switch (labelsView) {
-    case LabelView.ADD:
-      return (
-        <LabelAction
-          template={childTemplate}
-          label={undefined}
-          goBack={async () => {
-            setLabelsView(LabelView.LABELS);
-          }}
-          setTemplate={setChildTemplate}
-        />
-      );
-    case LabelView.MODIFY:
-      return (
-        <LabelAction
-          label={childTemplate.labels.filter((x) => x.reportTable === selectedLabel?.reportTable)[0]}
-          template={childTemplate}
-          goBack={async () => {
-            setLabelsView(LabelView.LABELS);
-          }}
-          setTemplate={setChildTemplate}
-        />
-      );
-    default:
-      return (
-        <>
-          <WidgetHeader
-            title={childTemplate.displayName === "" ? "Create template" : childTemplate.displayName}
-            disabled={isLoading}
-            returnFn={async () => {
-              await goBack();
+  return (
+    <>
+      <div className='ec3w-template-details-container' data-testid="ec3-template-details">
+        <Fieldset legend='Template Details' className='ec3w-template-details'>
+          <Small className='ec3w-template-field-legend'>
```

Comment:
It's a small thing, but maybe it would be good to create a `<RequiredFieldsNotice/>` component.

Reply:
Good idea. Done.